internal/k8s/pods/pods.go

```diff
@@ -40,10 +42,37 @@ func (pods *Pods) List(ctx context.Context, namespace string) ([]string, error)
 func (pods *Pods) KillByName(ctx context.Context, namespace string, podName string) error {
 	podsInNamespace := pods.client.CoreV1().Pods(namespace)
 	err := podsInNamespace.Delete(ctx, podName, v1.DeleteOptions{})
-
 	return err
 }
 
+// KillByKeyword kills the first pod with a name that contains the specified keyword
+func (pods *Pods) KillByKeyword(ctx context.Context, namespace string, podKeyword string) (string, error) {
+	// Iterate through podnames in list and find one that matches keyword
+	var podsList, err = pods.List(ctx, namespace)
+	var podToCheck = ""
+	for i := 0; i < len(podsList); i++ {
+		podToCheck = podsList[i]
+		if strings.Contains(podToCheck, podKeyword) {
+			break
+		}
+	}
+	var podName = podToCheck
+	err = pods.KillByName(ctx, namespace, podName)
+	return podName, err
+}
+
+// KillRandom kills a random pod within a namespace
+func (pods *Pods) KillRandom(ctx context.Context, namespace string) (string, error) {
+	// Randomly choose a pod from the given namespace
+	var podsList, err = pods.List(ctx, namespace)
+	var randomNum = rand.Intn(len(podsList) - 1)
+	var randomPod = podsList[randomNum]
+
+	// Kill the random pod by name
+	err = pods.KillByName(ctx, namespace, randomPod)
+	return randomPod, err
+}
+
```

Comment:
```suggestion
// KillByKeyword kills the first pod with a name that contains the specified keyword
func (pods *Pods) KillByKeyword(ctx context.Context, namespace string, podKeyword string) (string, error) {
	var podsList, err = pods.List(ctx, namespace)
	var podToCheck = ""
	for i := 0; i < len(podsList); i++ {
		podToCheck = podsList[i]
		if strings.Contains(podToCheck, podKeyword) {
			break
		}
	}
	var podName = podToCheck
	err = pods.KillByName(ctx, namespace, podName)
	return podName, err
}

// KillRandom kills a random pod within a namespace
func (pods *Pods) KillRandom(ctx context.Context, namespace string) (string, error) {
	var podsList, err = pods.List(ctx, namespace)
	var randomNum = rand.Intn(len(podsList) - 1)
	var randomPod = podsList[randomNum]

	err = pods.KillByName(ctx, namespace, randomPod)
	return randomPod, err
}

```

The intent is clearly expressed by the code itself, so no need for inline comments. 👍🏽